sql: fix call of `FindIndexWithName`

otan opened the pull request:
In e89093faf7225a635ddb5d4f260b20a4d8dd64e2, we added
`tableDesc.FindIndexWithName(name.String())`. However, `name.String()`
can return a `EncodeRestrictedSQLIdent` version of the string.

This fixes the call to use `string(name)`.

I couldn't produce an error with this at the moment, mostly because
there is some other check preventing it from happening from the
combinations I've tried.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/75115)
<!-- Reviewable:end -->

otan committed:
sql: fix call of `FindIndexWithName`

In e89093faf7225a635ddb5d4f260b20a4d8dd64e2, we added
`tableDesc.FindIndexWithName(name.String())`. However, `name.String()`
can return a `EncodeRestrictedSQLIdent` version of the string.

This fixes the call to use `string(name)`.

I couldn't produce an error with this at the moment, mostly because
there is some other check preventing it from happening from the
combinations I've tried.

Release note: None

postamar reviewed (approved):
I wonder how many other instances of this bug are out there. Is it worth making a linter check to prevent this? It's an easy mistake to make.

otan commented:
haha, yeah it's easy to make. linter? yes but i'm not 100% on the conditions to make it trigger...

bors r=postamar
